### PR TITLE
Dump genesis to JSON file

### PR DESCRIPTION
**Pull request by gavofyork**
Built on #217. Includes #226.

Use with `polkadot --build-genesis`. It'll stream to stdout. You can change the chain with `--chain=poc-2` or whatever as you like.

You can also read the file in the same format and it'll Just Work (since in Polkadot and Substrate chains generally, the genesis state is all that is needed to define the STF and thus the chain itself).

Use with `--chain=/some/filename/here.json`.

Also a bit of miscellaneous tidying; renamed `BuildExternalities` -> `BuildStorage` and `BuiltExternalities` -> `StorageMap`.

TODO:
- [x] Fix tests & build outside of `polkadot` binary.
- [x] Test.
- [x] Introduce poc-1 genesis state as a predefinition.

### CODE REVIEW

**Comment by gavofyork** [comment]
I think you misunderstand; this doesn't export the state of the network in order to configure a second genesis with a previous network's accrued state.

This rather exports the storage of the genesis block in order that a later version of polkadot, with a different runtime, can still bootstrap a previous genesis block. (This is only possible because Polkadot uses Substrate, which means the runtime is dynamically specified on-chain.)

**Comment by arkpar** [comment]
Still contains unintended changes to `substrate/executor` ?

File: substrate/executor/src/sandbox.rs [hunk 1]
```diff
@@ -113,26 +111,22 @@ impl ImportResolver for Imports {
...
+		_module_name: &str,
+		_field_name: &str,
 		_global_type: &::wasmi::GlobalDescriptor,
 	) -> Result<::wasmi::GlobalRef, ::wasmi::Error> {
-		Err(::wasmi::Error::Instantiation(format!(
-			"Export {}:{} not found",
-			module_name, field_name
-		)))
+		// TODO:
+		unimplemented!()
```
Comment (rphmeier) [inline substrate/executor/src/sandbox.rs]: returning an error is strictly better than panicking in this and `resolve_table` -- why were these implementations removed?

**Comment by gavofyork** [comment]
@arkpar @rphmeier fixed now

**Comment by arkpar** [comment]
Looks good, but needs another merge.

**Comment by gavofyork** [comment]
ain't it the truth...

**Pull request merged**